data_validation/result_handlers/factory.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Build a result handler object."""
+
+from typing import Union
+
+import google.oauth2.service_account
+
+from data_validation import consts, exceptions
+from data_validation.result_handlers.bigquery import BigQueryResultHandler
+from data_validation.result_handlers.postgres import PostgresResultHandler
+from data_validation.result_handlers.text import TextResultHandler
+
+
+def build_result_handler(
+    result_handler_config: dict,
+    validation_type: str,
+    filter_status: list,
+    text_format: str = None,
+) -> Union[TextResultHandler, BigQueryResultHandler]:
```

Comment:
```suggestion
) -> BaseBackendResultHandler:
```
By my sure understanding only one result handler is being returned at a time and since it can be any of them, the base class could be a better fit for the return type hints.

Reply:
Good spot, thanks